dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/notification/HibernateProgramNotificationInstanceStore.java

```diff
@@ -118,9 +124,30 @@ private List<Function<Root<ProgramNotificationInstance>, Predicate>> getPredicat
     }
 
     if (params.hasScheduledAt()) {
-      predicates.add(root -> builder.equal(root.get("scheduledAt"), params.getScheduledAt()));
+      Pair<Date, Date> scheduledAt = getStartAndEndOfDay(params.getScheduledAt());
+
+      predicates.add(
+          root ->
+              builder.and(
+                  builder.greaterThanOrEqualTo(root.get("scheduledAt"), scheduledAt.getLeft()),
+                  builder.lessThan(root.get("scheduledAt"), scheduledAt.getRight())));
     }
 
     return predicates;
   }
+
+  public Pair<Date, Date> getStartAndEndOfDay(Date scheduledAt) {
```

Comment:
Only for this branch. For higher branches, I will have a param mapper mapping `scheduledAt` to `scheduledFrom` and `scheduledTo`

Reply:
What do you mean by

> Only for this branch. For higher branches

Reply:
Normally, we handle this in the parameter mapper class. However, for version 2.39, I implemented it directly in the store. For higher versions, I plan to follow the standard approach using the parameter validation mapper class.